library/src/main/kotlin/com/connectrpc/protocols/ConnectInterceptor.kt

```diff
@@ -278,15 +279,10 @@ internal class ConnectInterceptor(
             if (detail.type == null) {
                 continue
             }
-            val payload = detail.value
-            if (payload == null) {
-                errorDetails.add(ConnectErrorDetail(detail.type, ByteString.EMPTY))
-                continue
-            }
             errorDetails.add(
                 ConnectErrorDetail(
                     detail.type,
-                    payload.encodeUtf8(),
+                    detail.value?.decodeBase64() ?: ByteString.EMPTY,
```

Comment:
This is basically the one place we need to base64-decode the error detail value. From here on out, they're decoded bytes.